superdesk/publish/publish_queue.py

```diff
@@ -126,6 +129,15 @@ def on_create(self, docs):
                 subscriber = subscriber_service.find_one(req=None, _id=doc['subscriber_id'])
                 doc['published_seq_num'] = subscriber_service.generate_sequence_number(subscriber)
 
+            # if keys are not returned by client (can happen on resend), we need to get them for database
+            keys = {'item_encoding', 'encoded_item_id'}
+            if not keys.issubset(doc):
+                item = get_resource_service("publish_queue").find_one(req=None, item_id=doc['item_id'])
+                if item is not None:
```

Comment:
There may be multiple entries in the publish queue with the same "item_id", If there are multiple format for example.